web-components/src/components/datepicker/DatePicker.ts

```diff
@@ -0,0 +1,123 @@
+import "@/components/datepicker/datepicker-calendar/DatePickerCalendar";
+import { addDays, addWeeks, DayFilters, isDayDisabled, now, subtractDays, subtractWeeks } from "@/utils/dateUtils";
+import { customElement, html, internalProperty, LitElement, property, query } from "lit-element";
+import { DateTime } from "luxon";
+import "../input/Input";
+import "../menu-overlay/MenuOverlay";
+import { MenuOverlay } from "../menu-overlay/MenuOverlay";
+
+export namespace DatePicker {}
+
+@customElement("md-datepicker")
+export class DatePicker extends LitElement {
+  @property({ type: Boolean }) shouldCloseOnSelect = true;
+  @property({ attribute: false }) locale: string = now().locale;
+  @property({ attribute: false }) monthFormat = undefined;
+  @property({ type: String }) maxDate: string | undefined = undefined;
+  @property({ type: String }) minDate: string | undefined = undefined;
+  @property({ attribute: false }) selectedDate: DateTime = now();
+  @property({ attribute: false }) focusedDate: DateTime = now();
+  @property({ attribute: false }) filterDate: Function | undefined = undefined;
+  @property({ attribute: false }) onChange: Function | undefined = undefined;
+  @property({ attribute: false }) onMonthChange: Function | undefined = undefined;
+  @property({ attribute: false }) onSelect: Function | undefined = undefined;
+
+  @internalProperty() maxDateData: DateTime | undefined = undefined;
+  @internalProperty() minDateData: DateTime | undefined = undefined;
+
+  @query("md-menu-overlay") menuOverlay!: MenuOverlay;
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.minDate !== undefined ? (this.minDateData = DateTime.fromSQL(this.minDate)) : null;
+    this.maxDate !== undefined ? (this.maxDateData = DateTime.fromSQL(this.maxDate)) : null;
+  }
+
+  setOpen = (open: boolean): void => {
+    this.menuOverlay.isOpen = open;
+  };
+
+  handleSelect = (e: CustomEvent): void => {
```

Comment:
Typescript can automatically infer types from function. Is this return typing really necessary ?

Reply:
Eh, probably not, I'm just on a Typescript kick!